File: Backend/src/main/java/tn/esprit/clubsync/entities/Book.java
package tn.esprit.clubsync.entities;


import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "books")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String author;

    private String isbn;

    private String publisher;

    private LocalDate publicationDate;

    private String genre;

    private String description;

    private String language;

    private Integer pageCount;

    private Boolean available = true;

    private Integer totalCopies;

    private Integer availableCopies;

    private String coverImageUrl;
}

File: dashboard/src/app/models/book.ts
export interface Book {
    id?: number;
    title: string;
    author: string;
    isbn?: string | null;      
    publisher?: string | null;
    publicationDate?: string | null; 
    genre?: string | null;
    description?: string | null;
    language?: string | null;
    pageCount?: number | null;
    available?: boolean;
    totalCopies?: number | null;
    availableCopies?: number | null;
    coverImageUrl?: string | null;
  }